### PR TITLE
Db logic delete

### PR DESCRIPTION
**Pull request by ncargnelutti104**

### CODE REVIEW

File: doc/gpsDataModel.sql [hunk 1]
```diff
@@ -16,20 +18,27 @@ create table USERS(
...
+	user_id serial references USERS(id) on delete restrict not null,
 	vehicle_id bigint unique references VEHICLES(id) on delete restrict not null,
 	model varchar(45) not null,
 	software_version varchar(45)
 );
 
 create table TRACKINGS(
 	id serial primary key not null,
-	device_id bigint references DEVICES(id) on delete restrict not null,
+	device_id serial references DEVICES(id) on delete restrict not null,
```
Comment (drhernandez) [inline doc/gpsDataModel.sql]: device id no es autoincrementable

File: src/main/java/com/tesis/services/imp/UserServiceImp.java [hunk 2]
```diff
@@ -71,7 +75,9 @@
     public ResponseDTO<Users> deleteUser(Integer userID) {
         ResponseDTO<Users> responseDTO = new ResponseDTO<>();
         try {
-            usersDao.deleteById(userID);
+            Users user = usersDao.fetchOneById(userID);
+            user.setDeletedAt(Timestamp.valueOf(LocalDateTime.now()));
```
Comment (drhernandez) [inline src/main/java/com/tesis/services/imp/UserServiceImp.java]:
```suggestion
            user.setDeletedAt(Timestamp.valueOf(LocalDateTime.now(Clock.systemUTC())));
```

File: src/main/java/com/tesis/services/imp/UserServiceImp.java [hunk 3]
```diff
@@ -49,6 +51,8 @@
     public ResponseDTO<Users> updateUser(Integer userID, Users newData) {
         ResponseDTO<Users> responseDTO = new ResponseDTO<>();
         Users user = usersDao.fetchOneById(userID);
+        user.setLastUpdated(Timestamp.valueOf(LocalDateTime.now()));
```
Comment (drhernandez) [inline src/main/java/com/tesis/services/imp/UserServiceImp.java]:
```suggestion
        user.setLastUpdated(Timestamp.valueOf(LocalDateTime.now(Clock.systemUTC())));
```

File: doc/gpsDataModel.sql (hunk 1 above)
Comment (ncargnelutti104) [reply doc/gpsDataModel.sql]: done

File: src/main/java/com/tesis/services/imp/UserServiceImp.java (hunk 2 above)
Comment (ncargnelutti104) [reply src/main/java/com/tesis/services/imp/UserServiceImp.java]: listo..

File: src/main/java/com/tesis/services/imp/UserServiceImp.java (hunk 3 above)
Comment (ncargnelutti104) [reply src/main/java/com/tesis/services/imp/UserServiceImp.java]: fatto

**Pull request merged**